### DISCUSSION TITLE
Trigger job on push on commit which is in a PR via if condition

### DISCUSSION THREAD

**Discussion by basickarl** [community/community#26356] [Actions]
<p>I seem to be unable to locate in the documentation where I can trigger a job with the following conditions only:</p>
<ol>
<li>on push</li>
<li>branch happens to be in a PR</li>
</ol>
<p>Which means that pushes on branches that are not currently in a PR would not trigger the job.</p>
<p>How would one write such a workflow yaml file?</p>
<p>I would like this to happen via the if: condition.</p>

**Comment by otter-computer** [comment]
<p>Hi <a class="mention" href="/basickarl">@basickarl</a>, welcome to the GitHub Support Community! I believe you should be able to do this without needing to use the <code>if</code> conditional, by listening to the <code>synchronize</code> type of the <code>pull_request</code> event:</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#pull_request" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="16" height="16">

<h3><a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#pull_request" target="_blank" rel="noopener">Events that trigger workflows - GitHub Docs</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<pre><code class="lang-yaml">on:
  pull_request:
    types: [synchronize]
</code></pre>
<p>Or is there a specific reason you need to use the <code>if</code> conditional? If so, it’d be helpful to hear why you’re listening for that!</p>

**Comment by basickarl** [comment]
<p>Hi thanks for the welcoming :slight_smile:.</p>
<p>I do wish to specifically do this via an if condition. I wish to do the following:</p>
<ol>
<li>build app</li>
<li>run tests</li>
<li>deploy if the commit push is in a PR.</li>
</ol>
<p>Basically I do not wish to deploy the application if it’s a normal push.</p>
<p>Something like this:</p>
<pre><code class="lang-auto">jobs:
    build:
        runs-on: ubuntu-latest
        strategy:
            matrix:
                node-version: [12.x]
        steps:
            - run: npm run build # &lt;--- runs on every push
            - run: npm run test # &lt;--- runs on every push
            - name: deploy # &lt;--- only runs if it is a PR
              if: pull_request === true (or something)
              run: ...my deploy code
</code></pre>

**Comment by otter-computer** [comment, marked as answer]
<p>Awesome, thanks for the context! So in this case you should be able to look at the <code>event_name</code> item inside the <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#github-context"><code>github</code> context</a> object.</p>
<pre><code class="lang-yaml">        steps:
            - name: deploy
              if: github.event_name == 'pull_request'
              run: ...
</code></pre>
<p>You could also specifically only listen for pushes to the pull request too:</p>
<pre><code class="lang-yaml">              if: github.event_name == 'pull_request' &amp;&amp; github.event.action == 'synchronize'
</code></pre>
<p>You can see the syntax for constructing these statements here:</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="16" height="16">

<h3><a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions" target="_blank" rel="noopener">Context and expression syntax for GitHub Actions - GitHub Docs</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by basickarl** [comment]
<p>Ah I see!</p>
<p>I did try the following:</p>
<pre><code class="lang-auto">name: Test

on:
    pull_request:
        types: [synchronize]

jobs:
    build:
        runs-on: ubuntu-latest
        strategy:
            matrix:
                node-version: [12.x]
        steps:
            - name: test
              if: github.event_name == 'pull_request' &amp;&amp; github.event.action == 'synchronize'
              run: echo hello
</code></pre>
<p>But it doesn’t seem to want to trigger a workflow now :frowning_face::</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/a/7/a7c030250ded56fa29b4617589fb78f3ff640bf6.png" data-download-href="/uploads/short-url/nVZuUBpweBEcnNLBhBLZQ1s0jDo.png?dl=1" title="Screenshot from 2020-11-30 19.02.18" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/10507098/181097311-2d6835db-f096-4016-8389-842f33935aa6.png" alt="Screenshot from 2020-11-30 19.02.18" data-base62-sha1="nVZuUBpweBEcnNLBhBLZQ1s0jDo" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/a/7/a7c030250ded56fa29b4617589fb78f3ff640bf6_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/3X/a/7/a7c030250ded56fa29b4617589fb78f3ff640bf6_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/a/7/a7c030250ded56fa29b4617589fb78f3ff640bf6.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/7/a7c030250ded56fa29b4617589fb78f3ff640bf6_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot from 2020-11-30 19.02.18</span><span class="informations">1096×548 25 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I am force pushing from branch km/auth0 (the workflow file is in that branch).</p>

**Comment by basickarl** [comment]
<p>I solved the issue. It was that the PR had a merge conflict with the master branch. I resolved it and now the pull_request event is triggering correctly. Thanks <a class="mention" href="/thomasshaped">@thomasshaped</a> for your code, it also helped!</p>
